cinn/auto_schedule/measure/measurer_test.cc

```diff
@@ -44,34 +44,76 @@ frontend::Program CreateAddReluProgram() {
   return builder.Build();
 }
 
-TEST(ScheduleMeasurer, Basic) {
+class TestMeasurer : public ::testing::Test {
+ public:
+  std::unique_ptr<GraphCompiler> graph_compiler;
+  std::vector<TuneTask> tasks;
+  std::vector<MeasureInput> inputs;
+
+  void SetUp() override {
 #ifdef CINN_WITH_CUDA
-  Target target = common::DefaultNVGPUTarget();
+    Target target = common::DefaultNVGPUTarget();
 #else
-  Target target = common::DefaultHostTarget();
+    Target target = common::DefaultHostTarget();
 #endif
-  auto graph          = std::make_shared<Graph>(CreateAddReluProgram(), target);
-  auto scope          = BuildScope(target, graph);
-  auto graph_compiler = std::make_unique<GraphCompiler>(target, scope, graph);
-  TaskCreator task_creator;
-  std::vector<TuneTask> tasks = task_creator.CreateTuneTaskOpLevel(graph.get());
-  ASSERT_EQ(2, tasks.size());
-
-  std::vector<MeasureInput> inputs(tasks.size());
-  for (int i = 0; i < tasks.size(); ++i) {
-    auto* task     = &tasks[i];
-    inputs[i].task = task;
-    // TODO(CtfGo): currently FusedGraphToLoweredFunc doesn't work well on NVGPU target,
-    // this setting of lowered_funcs will be enabled once we fix the bug
-    // inputs[i].lowered_funcs = graph_compiler->FusedGraphToLoweredFunc(task->task_graph());
+    auto graph     = std::make_shared<Graph>(CreateAddReluProgram(), target);
+    auto scope     = BuildScope(target, graph);
+    graph_compiler = std::make_unique<GraphCompiler>(target, scope, graph);
+    TaskCreator task_creator;
+    tasks = task_creator.CreateTuneTaskOpLevel(graph.get());
+    inputs.reserve(tasks.size());
+    for (int i = 0; i < tasks.size(); ++i) {
+      auto* task = &tasks[i];
+      MeasureInput input;
+      input.task = task;
+      // TODO(CtfGo): currently FusedGraphToLoweredFunc doesn't work well on NVGPU target,
+      // this setting of lowered_funcs will be enabled once we fix the bug
+      // input.lowered_funcs = graph_compiler->FusedGraphToLoweredFunc(task->task_graph);
+      inputs.emplace_back(input);
+    }
   }
+};
+
+class ThrowExceptionBuilder : public ScheduleBuilder {
+  struct Exception : public std::exception {
+    const char* what() const throw() { return "BuildError"; }
+  };
+  BuildResult Build(const MeasureInput& input) override { throw Exception(); }
+};
 
+class ThrowExceptionRunner : public ScheduleRunner {
+  struct Exception : public std::exception {
+    const char* what() const throw() { return "RunError"; }
+  };
+  MeasureResult Run(const MeasureInput& input, const BuildResult& build_result) override { throw Exception(); }
+};
+
+TEST_F(TestMeasurer, Basic) {
```

Comment:
Should we test a successful case that `ScheduleMeasurer`'s number of threads is 1, to make sure it can work in single thread. In the future, we may debug on single thread for sometimes when we are not sure where the error happens.

Reply:
Good